src/plugins/workspace/public/plugin.ts

```diff
@@ -118,6 +123,13 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
             currentAppIdSubscription.unsubscribe();
           });
         })();
+
+        // If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu and disable
```

Comment:
Nit: better to extract to a private method and listen the currentWorkspaceId$ change. `currentWorkspaceId$` should be the single source of truth for detecting if is inside a workspace.